processing/src/test/java/io/druid/granularity/QueryGranularityTest.java

```diff
@@ -746,4 +748,15 @@ private void assertSame(List<DateTime> expected, Iterable<Long> actual)
     Assert.assertFalse("actualIter not exhausted!?", actualIter.hasNext());
     Assert.assertFalse("expectedIter not exhausted!?", expectedIter.hasNext());
   }
+  
+  @Test(timeout = 10_000L)
+  public void testDeadLock() throws Exception
+  {
+    final URL[] urls = ((URLClassLoader)QueryGranularity.class.getClassLoader()).getURLs();
+    final String className = QueryGranularity.class.getCanonicalName();
+    for(int i = 0; i < 1000; ++i) {
```

Comment:
@drcrallen do you remember why you added 1000 iterations here? Was it necessary to catch a deadlock?